deployment/ccip/changeset/cs_promote_candidate_test.go

```diff
@@ -0,0 +1,142 @@
+package changeset
+
+import (
+	"testing"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/smartcontractkit/chainlink-testing-framework/lib/utils/testcontext"
+	"github.com/smartcontractkit/chainlink/deployment"
+	"github.com/smartcontractkit/chainlink/deployment/ccip/changeset/internal"
+	commonchangeset "github.com/smartcontractkit/chainlink/deployment/common/changeset"
+	"github.com/smartcontractkit/chainlink/deployment/environment/memory"
+	"github.com/smartcontractkit/chainlink/v2/core/capabilities/ccip/types"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/exp/maps"
+)
+
+func Test_PromoteCandidate(t *testing.T) {
```

Comment:
can we keep it in cs_active_candidate_test so its clear where all the tests for given changesets live

Reply:
IMO cs_active_candidate.go is kinda confusing, having them in separate files w/ separate unit tests & configs is better? Don't feel too strongly about it tho

Reply:
I think the main thing is making sure that cs_x.go has associated tests in cs_x_test.go so they are easy to find. Then imo helpful to bundle all the related changesets to CCIPHome in one file, especially since they are unused in conjunction w each other. Maybe instead of cs_active_candidate we call it cs_ccip_home ?

Reply:
Alright, will do for now, though I think for more granular coverage I'd actually like this to be in a subpackage within changesets and each changeset have its own unit test

Reply:
each cs should totally have its own unit test, I'm just talking about files

Reply:
Yeah, makes sense to me.